src/utils/dockerContextManager.ts

```diff
@@ -0,0 +1,190 @@
+                        // Config file will change when Docker context is changed, but opposite is not necessarily true
+                        // (i.e. config file might change for other reasons).
+                        await this.refreshCachedDockerContext();
+                    }
+                }
+            } finally {
+                // Start counting time _after_ we are done with all the I/O
+                this.lastContextCheckTimestamp = Date.now();
+            }
+        }
+
+        return {
+            Context: this.cachedContext,
+            Changed: this.contextChanged
+        };
+    }
+
+    public contextChangeHandled(): void {
+        this.contextChanged = false;
+    }
+
+    private async getDockerConfigDigest(): Promise<string> {
+        // Note: computing Docker config file digest may fail, typically because Docker is not installed,
+        // and there is no config file. We use falsy value (empty string) as a way to indicate that,
+        // as opposed to rejecting the promise with the captured error.
+        // This is a clue to the caller that they should go ahead and try to inspect the actual Docker context.
+
+        return new Promise<string>((resolve, reject) => {
+            try {
+                // tslint:disable-next-line: non-literal-fs-path
+                const dockerConfig = fs.createReadStream(DockerConfigFilePath);
+                let hash = crypto.createHash('sha256');
+                hash = dockerConfig.pipe(hash, { end: false });
+
+                dockerConfig.on('end', () => {
+                    try {
+                        const digest = hash.digest('hex');
+                        resolve(digest);
+                    } catch (ex) {
+                        resolve('');
+                    }
+                });
+
+                dockerConfig.on('error', _ => {
+                    resolve('');
+                })
+            } catch (ex) {
+                resolve('');
+            }
+        });
+    }
+
+    private async refreshCachedDockerContext(): Promise<void> {
+        const { Result: currentContext, DurationMs: duration } = await timeUtils.timeIt(async () => this.inspectCurrentContext());
+
+        if (!this.cachedContext || currentContext.FullSpec !== this.cachedContext.FullSpec) {
+            const previousContext = this.cachedContext;
+            this.contextChanged = true;
+            this.cachedContext = currentContext;
+            this.sendDockerContextEvent(currentContext, previousContext, duration);
+        }
+    }
+
+    private async inspectCurrentContext(): Promise<IDockerContext> {
+        let execResult: {
+            stdout: string;
+        };
+        let contextOverride: string = process.env[DOCKER_CONTEXT] || '';
+        if (contextOverride) {
+            contextOverride = ' ' + contextOverride;
+        }
+        const inspectCmd = 'docker context inspect' + contextOverride;
```

Comment:
```suggestion
        const inspectCmd = `docker context inspect ${process.env[DOCKER_CONTEXT] || ''}`;
```

Then can get rid of `contextOverride`.

Reply:
Nice

Reply:
String interpolation is basically the best thing ever in every language.